konrad/aerosol.py

```diff
@@ -0,0 +1,289 @@
+                attrs={'units': units},
+            ).T
+
+    raise TypeError(
+        'Aerosol variable input must be a single value, `numpy.ndarray` or a '
+        '`sympl.DataArray`')
+
+class Aerosol(metaclass=abc.ABCMeta):
+    def __init__(self,atmNumlevels, aerosol_type='no_aerosol', aerosolLevelShiftInput=0,monthsAfterEruption=2,includeSWForcing=True,\
+                 includeLWForcing=True,includeScattering=True,includeAbsorption=True):
+        self._aerosol_type = aerosol_type
+        self.includeSWForcing=includeSWForcing
+        self.includeLWForcing=includeLWForcing
+        self.aerosolLevelShift=aerosolLevelShiftInput
+        self.includeScattering=includeScattering
+        self.includeAbsorption=includeAbsorption
+        self.optical_thickness_due_to_aerosol_sw = \
+            get_aerosol_waveband_data_array(0, units='dimensionless', numlevels=atmNumlevels, sw=True)   #called ext_sun in files
+        self.single_scattering_albedo_aerosol_sw = \
+            get_aerosol_waveband_data_array(0, units='dimensionless', numlevels=atmNumlevels, sw=True) #called omega_sun in files
+        self.asymmetry_factor_aerosol_sw =  \
+            get_aerosol_waveband_data_array(0, units='dimensionless', numlevels=atmNumlevels, sw=True)  #called g_sun in files
+        self.optical_thickness_due_to_aerosol_lw = \
+            get_aerosol_waveband_data_array(0, units='dimensionless', numlevels=atmNumlevels, sw=False) #called ext_earth in files
+
+
+    #################################################################
+    #To do: 
+    #- time step updating
+    #  For now the aerosols are left constant and are not updated
+    #- implementation for a changing lapse rate, for now it is implemented only for a fixed lapse rate
+    #- add the cold point as an output
+    ################################################################
+    
+    
+    def update_aerosols(self, time, atmosphere):
+        return
+    
+
+class VolcanoAerosol(Aerosol):
+    '''
+    CMIP6 volcanic aerosols (Mount Pinatubo eruption)
+    '''
+    def __init__(self,atmNumlevels, aerosolLevelShiftInput=0,includeSWForcing=True,\
+                 includeLWForcing=True,includeScattering=True,includeAbsorption=True):
+        super().__init__(atmNumlevels,aerosol_type='all_aerosol_properties') 
+        self.aerosolLevelShift=aerosolLevelShiftInput
+        self.numlevels=atmNumlevels
+        self.includeSWForcing=includeSWForcing
+        self.includeLWForcing=includeLWForcing
+        self.includeScattering=includeScattering
+        self.includeAbsorption=includeAbsorption
+
+    def update_aerosols(self, time, atmosphere):
+        '''
+        Import the volcanic forcing data in first time step, taking into account
+        the parameter settings of the VolcanoAerosol class.
+        Translate from height dependance to pressure dependance.
+        The aerosol layer is kept fixed and constant throughout the following run.
+        '''
```

Comment:
The docstring (or the one in the parent class) should explain the input arguments.